book/src/juice.md

```diff
@@ -0,0 +1,43 @@
+
+To those types there’s three parameters: magnitude, frequency and length. Magnitude affects how far the screen moves when shaking (the amplitude of the oscillation) while frequency controls how ”many” shakes happen per second. The length decides for how long the shake will last for, and the magnitude decreases over this time to zero.
+
+By mixing and matching different values for these parameters different effects can be made. Here are a few examples:
+- Gunshot: a short omnidirectional shake with low amplitude and high frequency. This gives a small kick that doesn’t move the screen too much, just enough to confirm the firing.
+Fish Fight supports setting multipliers for X/Y movement, so a 1 for X and a smaller number for Y makes the player feel the direction of the action.
+- Explosion: a medium-length omnidirectional shake with decent amplitude and lower frequency. This makes a kind of rumble that seriously jostles the screen. The bigger the explosive the lower the frequency!
+- Jump impact: a directional sinusoidal shake makes the world bounce like a spring. 
+
+To calculate the final scene position, the current position of every active shake is added together. That value is then capped using a logarithmic function so that if many explosives go off at once, the scene does not fly off-screen. The second logarithm (log<sub>2</sub>x) is useful for this, as it is almost linear from 0-1 and then quickly flattens.
+
+## Particles
+Particles like those in games rarely occur in real life, as most of us go through our daily routine avoiding high-impact situations. But games tend to be a bit more forceful and violent, and that of course means heaps of particles! Particles are a great way to create dynamic and exciting effects that further give impact and weight to actions, but also make the world seem more alive and responsive. Some examples are sparks flying when a bullet hits a wall or dust blowing around in a storm. 
+
+In Fish Fight, [macroquad-particles](https://crates.io/crates/macroquad-particles) submodule is used to handle particles. While it might look rudimentary, it is deceptively so. With several particles, blending modes and curse pretty impressive effects can be made. The explosions in FishFight used to be entirely procedural! Now it is used for muzzle smoke, trails, bubbles and whatever other details will be added in the future. 
+
+Particle presets are stored in .json files, which can be either generated manually or using [this excellent editor](https://fedorgames.itch.io/macroquad-particles) which provides previews and live tweaking. It might not be entirely up to date with the latest particle system format, but should get you up and running quickly.
```

Comment:
I really like this section. I wasn't sure how we were finding those values in the particle json files, so this helps me understand this part of the codebase a bit more.

Reply:
It only took a bunch of "find all references" clicks and a while... I can't guarantee it's right but the others are saying lgtm 👍